main/ZgatewayRTL_433.ino

```diff
@@ -0,0 +1,135 @@
+
+    OpenMQTTGateway is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "User_config.h"
+
+#ifdef ZgatewayRTL_433
+
+#  include <rtl_433_ESP.h>
+
+#  define CC1101_FREQUENCY 433.92
+#  define JSON_MSG_BUFFER  512
+#  define ONBOARD_LED      2
+
+char messageBuffer[JSON_MSG_BUFFER];
+
+rtl_433_ESP rtl_433(-1); // use -1 to disable transmitter
+
+#  include <ELECHOUSE_CC1101_SRC_DRV.h>
+
+void rtl_433_Callback(char* message) {
+  DynamicJsonBuffer jsonBuffer2(JSON_MSG_BUFFER);
+  JsonObject& RFrtl_433_ESPdata = jsonBuffer2.parseObject(message);
+
+  pub(subjectRTL_433toMQTT, RFrtl_433_ESPdata);
+#  ifdef MEMORY_DEBUG
+  Log.trace(F("Post rtl_433_Callback: %d" CR), ESP.getFreeHeap());
+#  endif
+}
+
+void rtl_433setup() {
+  rtl_433.initReceiver(RF_RECEIVER_GPIO, receiveMhz);
+  rtl_433.setCallback(rtl_433_Callback, messageBuffer, JSON_MSG_BUFFER);
+  Log.notice(F("ZgatewayRTL_433 command topic: %s%s" CR), mqtt_topic, subjectMQTTtoRTL_433);
+  enableRTLreceive();
+  Log.trace(F("ZgatewayRTL_433 setup done " CR));
+}
+
+void rtl_433loop() {
+  rtl_433.loop();
+}
+
+extern void MQTTtoRTL_433(char* topicOri, JsonObject& RTLdata) {
+  if (cmpToMainTopic(topicOri, subjectMQTTtoRTL_433)) {
+    Log.trace(F("MQTTtoRTL_433 %s" CR), topicOri);
+    float tempMhz = RTLdata["mhz"];
+    int minimumRssi = RTLdata["rssi"] | 0;
+    int debug = RTLdata["debug"] | -1;
+    int status = RTLdata["status"] | -1;
+    if (tempMhz != 0 && validFrequency((int)tempMhz)) {
+      //  activeReceiver = RTL; // Enable RTL_433 Gateway
+      receiveMhz = tempMhz;
+      Log.notice(F("RTL_433 Receive mhz: %F" CR), receiveMhz);
+      pub(subjectRTL_433toMQTT, RTLdata);
+    } else if (minimumRssi != 0) {
+      Log.notice(F("RTL_433 minimum RSSI: %d" CR), minimumRssi);
+      rtl_433.setMinimumRSSI(minimumRssi);
+      pub(subjectRTL_433toMQTT, RTLdata);
+    } else if (debug >= 0 && debug <= 4) {
+      Log.notice(F("RTL_433 set debug: %d" CR), debug);
+      rtl_433.setDebug(debug);
+      rtl_433.initReceiver(RF_RECEIVER_GPIO, receiveMhz);
+      pub(subjectRTL_433toMQTT, RTLdata);
+    } else if (status >= 0) {
+      Log.notice(F("RTL_433 get status: %d" CR), status);
+      rtl_433.getStatus(status);
+      pub(subjectRTL_433toMQTT, RTLdata);
+    } else {
+      pub(subjectRTL_433toMQTT, "{\"Status\": \"Error\"}"); // Fail feedback
+      Log.error(F("MQTTtoRTL_433 Fail json" CR));
+    }
+    // enableActiveReceiver();
+  }
+}
+
+extern void enableRTLreceive() {
+  Log.trace(F("enableRTLreceive: %F" CR), receiveMhz);
+#  ifdef ZgatewayRF
+  disableRFReceive();
+#  endif
+#  ifdef ZgatewayPilight
+  disablePilightReceive();
```

Comment:
I think we are missing these 2 methods on the other gateways

Reply:
Correct, that's a carry over from my other WIP on switching active receiver.  Will remove until I can publish that feature.  PS I was using it during testing to compare PiLight with RTL_433